bibtexparser/middlewares/middleware.py

```diff
@@ -74,7 +74,18 @@ def metadata_key() -> str:
     # docstr-coverage: inherited
     def transform(self, library: "Library") -> "Library":
         # TODO Multiprocessing (only for large library and if allow_multi..)
-        blocks = [self.transform_block(b, library) for b in library.blocks]
+        blocks = []
+        for b in library.blocks:
+            transformed = self.transform_block(b, library)
+            # Case 1: None. Skip it.
+            if transformed is None:
+                pass
+            # Case 2: A single block. Add it to the list.
+            elif isinstance(transformed, Block):
+                blocks.append(transformed)
+            # Case 3: Something else. Assume it's a collection and append them all.
+            else:
```

Comment:
I am considering if we should now use the chance to double-check if "something else" is really a collection of blocks, and throw an exception otherwise. I guess that would be a relatively easy and runtime-wise cheap check, but heavily simplify debugging if any (e.g. custom) middleware returns something unexpected. 

What do you think?

Reply:
I think it's a good idea. I'll make the change.